blueocean-commons/src/main/java/io/jenkins/blueocean/commons/stapler/TreeResponse.java

```diff
@@ -31,14 +34,42 @@
         public Object invoke(StaplerRequest request, StaplerResponse response, Object instance, Object[] arguments)
                 throws IllegalAccessException, InvocationTargetException, ServletException {
 
-            final Object resp = target.invoke(request, response, instance, arguments);
+            /**
+             *  If request.method and HTTP verb annotations {@link GET}, {@link POST}, {@link PUT} and {@link DELETE}
+             *  do not match it skips invoking this target. If there no such annotations present then GET as default is
+             *  assumed and request is dispatched to target.
+             */
+            if (matches(request)) {
```

Comment:
@michaelneale It was a bug that was hiding, adding a DELETE operation exposed it. Without this if you have multiple methods operating at '/' routing level can be selected arbitrarily. This enforces method level routing based on HTTP method annotation, thats the intent anyways.